scripts/fix_sq_headers.py

```diff
@@ -0,0 +1,229 @@
+        return None  # Probably file not found
+
+    m5 = {}
+    for line in run.stdout.splitlines():
+        if line.startswith('@SQ'):
+            field = parse_one_SQ(line)
+            m5[field['SN']] = field['M5']
+
+    text = '/'.join([f'{name}:{m5[name]}' for name in sorted(m5.keys())])
+    hashed = hashlib.md5(text.encode('ASCII')).hexdigest()
+
+    return KNOWN_REFS.get(hashed, hashed)
+
+
+def do_reheader(dry_run, refset, newref, newrefset, incram, outcram, outcrai):
+    ref = classify_SQ(incram)
+    if ref != refset:
+        raise ValueError(f'{incram}: unexpected original reference headers: {ref}')
+
+    translate = {}
+    with open(newref) as refp:
+        for line in refp:
+            if line.startswith('@SQ'):
+                field = parse_one_SQ(line)
+                translate[field['SN']] = (line.strip(), field['LN'])
+
+    head_cmd = ['samtools', 'head', incram]
+    headers = sb.run(head_cmd, check=True, capture_output=True, text=True).stdout
+
+    newhdr_file = os.path.join(os.environ['BATCH_TMPDIR'], 'tmp.hdr')
+    with open(newhdr_file, 'w') as newhdr:
+        for line in headers.splitlines():
+            if line.startswith('@SQ'):
+                field = parse_one_SQ(line)
+                new_line, new_len = translate[field['SN']]
+                if field['LN'] != new_len:
+                    raise ValueError(f'{incram}:{field["SN"]}: new length {new_len} differs '
+                                     f'from existing {field["LN"]}')
+                print(new_line, file=newhdr)
+            else:
+                print(line, file=newhdr)
+
+    if dry_run:
+        print(f'{incram} original headers: {ref}')
+        print(f'Would rewrite it to {outcram} with headers:')
+        with open(newhdr_file) as fp:
+            print(fp.read())
+        return
+
+    sb.run(['samtools', 'reheader', '--no-PG', '--in-place', newhdr_file, incram], check=True)
+    os.rename(incram, outcram)
+    sb.run(['samtools', 'index', '-o', outcrai, outcram], check=True)
+
+    newref = classify_SQ(outcram)
+    print(f'{outcram}: reheadered: {newref}')
+    if newrefset is not None and newref != newrefset:
+        raise ValueError(f'{outcram}: unexpected reference headers written: {newref}')
+
+    return {
+        'cram_size': os.stat(outcram).st_size,
+    }
+
+
+def do_metamist_update(dataset, sequencing_type, seqgroup, oldpath, newpath, result):
+    analysis = models.Analysis(
+        type='cram',
+        status=models.AnalysisStatus('completed'),
+        output=str(newpath),
+        sequencing_group_ids=[seqgroup],
+        meta={
+            'sequencing_type': sequencing_type,
+            'size': result['cram_size'],
+            'source': f'Reheadered from {oldpath}',
+        },
+    )
+
+    aid = AnalysisApi().create_analysis(dataset, analysis)
+    print(f'Created Analysis(id={aid}, output={newpath}) in {dataset}')
+
+
+if __name__ == '__main__':
+    config = get_config(True)
```

Comment:
Nitpick, can you wrap this in a function that gets called with the args?

Reply:
Done, as a byproduct of converting to use command-line options via click.